### PR TITLE
[Merged by Bors] - Decode user id in path only in route

### PR DESCRIPTION
**Pull request by acrrd**
The entity `UserId` can be constructed from any string that is non-empty and does not contain the null byte.

This PR move the decoding of the URL parameter in the route definition.

### CODE REVIEW

**Review by Robert-Steiner — APPROVED** [review]
👍

**Review by biernacki — APPROVED** [review]
good change 👍

File: discovery_engine_core/web-api/src/routes.rs [hunk 1]
```diff
@@ -53,7 +57,14 @@ fn delete_internal_state(db: Db) -> impl Filter<Extract = impl Reply, Error = Re
...
 // PATH /user/:user_id
 fn user_path() -> impl Filter<Extract = (UserId,), Error = Rejection> + Clone {
-    warp::path("user").and(warp::path::param::<UserId>())
+    warp::path("user")
+        .and(warp::path::param::<String>())
+        .and_then(|user_id: String| async move {
+            urlencoding::decode(&user_id)
+                .map_err(Error::UserIdUtf8Conversion)
+                .and_then(UserId::new)
+                .map_err(|_| warp::reject())
```
Comment (rustonaut) [inline discovery_engine_core/web-api/src/routes.rs:66]: This produces a `404` instead of a `400`.

Comment (acrrd) [reply discovery_engine_core/web-api/src/routes.rs]: I changed this to return the error from `UserId::new`. That can be handled correctly in the task we have for the errors.

**Pull request closed without merging**